clients/client-java/src/main/java/com/datastrato/gravitino/client/api/SupportsCatalogs.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2023 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.client.api;
+
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.CatalogChange;
+import com.datastrato.gravitino.CatalogProvider;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.annotation.Evolving;
+import com.datastrato.gravitino.exceptions.CatalogAlreadyExistsException;
+import com.datastrato.gravitino.exceptions.NoSuchCatalogException;
+import com.datastrato.gravitino.exceptions.NoSuchMetalakeException;
+import java.util.Map;
+
+/**
+ * Interface for supporting catalogs. It includes methods for listing, loading, creating, altering
```

Comment:
It would be better to add some explanation about the difference between `SupportsCatalogs`  and that in the server side.

Reply:
added